packages/rolldown/tests/fixtures/tree-shake/external-export-star1/_config.ts

```diff
@@ -0,0 +1,20 @@
+import type { RolldownOutputChunk } from 'rolldown'
+import { defineTest } from '@tests'
+import { expect } from 'vitest'
+
+export default defineTest({
+  config: {
+    treeshake: {
+      moduleSideEffects: false,
+    },
+    external: ['fs'],
+  },
+  afterTest: (output) => {
+    output.output
+      .filter(({ type }) => type === 'chunk')
+      .forEach((chunk) => {
+        let code = (chunk as RolldownOutputChunk).code
+        expect(code.includes(`import * as import_fs from "fs"`))
```

Comment:
What's the blocker for writing this test in rust?

Reply:
https://github.com/rolldown/rolldown/pull/1558#discussion_r1670548422, just personal preference. Also, update I would not like to add too many snapshots because it would be some nightmare when you have some changes on codegen.

Reply:
This looks like manually snapshot. Beyond personal preferences, rust side test is easily to be debugging and has faster development feedback considering no need to compile the binding, so we should always prefer rust side tests if possible.

Tests of nodejs is only for testing API.

Reply:
https://github.com/rolldown/rolldown/blob/a3ada216a0d8a3404bdfde5feeb18c7c07a18f0c/crates/rolldown_common/src/inner_bundler_options/types/treeshake.rs#L3-L7, the deserialize is not easy to write for `treeshake`(especially when we fully support the configuration it may have function), at that moment, we will have half of the tree shake option related test in rust side, and another half in js side, so I still prefer writing test case in js

Reply:
Yeah. I understand your preferences.

> Tests of nodejs is only for testing API or config that can't represent in pure rust.

However, this is the intension. If there aren't blockers, please write pure rust test first.

Reply:
I also need it. see https://github.com/rolldown/rolldown/pull/1681